### ISSUE TITLE
"showscale=False" not respected for heatmap via update_traces

### ISSUE THREAD

**Issue by neili02** [plotly/plotly.py#1971]
For a gray scale image created via plotly.express, updating the trace to remove the colorbar does not work.

`im = np.random.randint(256,size=(50,50),dtype=np.uint8)`
 `fig = px.imshow(im, color_continuous_scale='gray')`
 `fig.update_traces(showscale=False)`
 `fig.show()`
I would expect this code to show the image without the colorbar but this is not the case.
Also, updating any colorbar property in this way does not work.

If the image plot is created using go.Heatmap(...) updating to `showscale=False` does work.

**Comment by emmanuelle** [comment]
Hi @neili02 it is because `px.imshow` uses a coloraxis for the colorbar, so the code should be
```
im = np.random.randint(256,size=(50,50),dtype=np.uint8)
fig = px.imshow(im, color_continuous_scale='gray')
fig.update_layout(coloraxis_showscale=False)
fig.show()
```
(you can see that a coloraxis is used by doing `print(fig)`). But we should update the documentation with an example showing how to remove the colorbar, so thank you for the report.

**Comment by neili02** [comment]
Thanks a lot for that info, the update_layout code does work.

**Comment by emmanuelle** [comment]
Closed by #1975. Thank you @neili02 !
